### PR TITLE
Refactor pre commit hook

### PR DESCRIPTION
**Pull request by manuelmessner**
Added whitespace protection and simplified some parts

### CODE REVIEW

**Comment by neithernut** [comment]
ACK.

**Comment by matthiasbeyer** [comment]
Tell me how to run a syntax check on the `pre-commit` file.

**Comment by manuelmessner** [comment]
You can "execute" the script by hand with the `-n` flag.
sh -n ./pre_commit

From man set (man sh links to this man page):

>  −n    The shell shall read commands but does not execute them; this can be used to check for shell script syntax errors. An interactive shell may ignore this option

**Comment by matthiasbeyer** [comment]
So this seems to be valid.

@manuelmessner are you ready for merge?

**Comment by neithernut** [comment]
We are evaluating the possibility of checking whether to use color output by querying git.
If we do not find a good way in the next few minutes, this may be merged.

**Comment by neithernut** [comment]
@manuelmessner found a way to query the effective color setup. He will integrate this into the hook.

**Comment by neithernut** [comment]
ACK.

File: hooks/pre-commit [hunk 1]
```diff
@@ -10,13 +10,20 @@ do
...
-cd $(git rev-parse --show-toplevel)
-
-Green="\033[32m"
-Red="\033[31m"
-Cyan="\033[0;36m"
-Color_Off="\033[0m"
+p="$(pwd)"
+cd "$(git rev-parse --show-toplevel)"
+
+if git config --get-colorbool color.ui; then
```
Comment (matthiasbeyer) [inline hooks/pre-commit:18]: I would like to have a short comment on this line, why it works as expected (for documentation).

**Comment by neithernut** [comment]
Again: ACK.

**Pull request merged**